### PR TITLE
refactor(src/tasks/package.js): simplify scripts

### PR DESCRIPTION
**Pull request by michael-ciniawsky**
#### Scripts

- 'clean:dist':  =>  'clean'

#### Scripts

- 'build:watch': use babel watch 

#### Scripts

- 'start'
- 'serve:dev'

#### devDependencies

- nodemon

### CODE REVIEW

File: src/tasks/package.js [hunk 1]
```diff
@@ -29,32 +29,33 @@ const packages = [
...
         // Some versions are skipped because of known issues, see https://github.com/webpack-contrib/organization/issues/7
         node: `>= ${config.minNode} < 5.0.0 || >= 5.10`,
       },
       scripts: {
-        start: 'yarn run serve:dev src',
-        build: "cross-env NODE_ENV=production babel src -d dist --ignore 'src/**/*.test.js'",
-        'clean:dist': 'del-cli dist',
-        lint: 'eslint --cache src test',
+        start: 'yarn run build -- -w',
+        build: 'cross-env NODE_ENV=production babel src -d dist',
```
Comment (d3viant0ne) [inline src/tasks/package.js]: Last this was discussed, we aren't requiring tests to exist in a `test` folder.

File: src/tasks/package.js [hunk 2]
```diff
@@ -29,32 +29,33 @@ const packages = [
...
+        'clean': 'del-cli dist test/builds',
+        lint: 'eslint --cache src',
```
Comment (d3viant0ne) [inline src/tasks/package.js]: Linting tests is a generally accepted practice, why was this removed?

File: src/tasks/package.js [hunk 3]
```diff
@@ -29,32 +29,33 @@ const packages = [
...
-        prebuild: 'yarn run clean:dist',
+        prebuild: 'yarn run clean',
         prepublish: 'yarn run build',
         release: 'yarn run standard-version',
         security: 'nsp check',
-        'serve:dev': 'nodemon $2 --exec babel-node',
-        test: 'jest',
+        develop: 'nodemon $2 --exec babel-node',
+        pretest: 'yarn run build && node test/webpack.build.js',
+        test: 'jest --verbose',
```
Comment (d3viant0ne) [inline src/tasks/package.js]: Shouldn't be `verbose` by default. Someone can always execute with `npm run test -- --verbose` if they want the extra output.

File: src/tasks/package.js [hunk 4]
```diff
@@ -29,32 +29,33 @@ const packages = [
...
         'lint-staged': 'lint-staged',
-        prebuild: 'yarn run clean:dist',
+        prebuild: 'yarn run clean',
         prepublish: 'yarn run build',
         release: 'yarn run standard-version',
         security: 'nsp check',
-        'serve:dev': 'nodemon $2 --exec babel-node',
-        test: 'jest',
+        develop: 'nodemon $2 --exec babel-node',
+        pretest: 'yarn run build && node test/webpack.build.js',
```
Comment (d3viant0ne) [inline src/tasks/package.js]: Jest doesn't require to execute a build, to execute a test. `babel-jest` takes care of this & was one of the reasons why we went with jest. Building on every test run execution is less than optimal.

**Comment by d3viant0ne** [comment]
I don't personally think it's practical to try and enforce a test configuration at this level of detail.

The use cases across the current & future contrib repos are just to broad for something like this to be a 'default'. Documented examples somewhere or even an example loader repo would be just as useful and significantly less limiting.

File: src/tasks/package.js (hunk 4 above)
Comment (michael-ciniawsky) [inline src/tasks/package.js]: When running 'through' webpack it needs to build the src && build the tests with webpack => Jest tests the bundle afterwards, but yeah maybe better for a separate PR

File: src/tasks/package.js (hunk 2 above)
Comment (michael-ciniawsky) [inline src/tasks/package.js]: I thought test linting will be skipped following up on the suggestion you made @coffee-loader, that's why I removed it by default

File: src/tasks/package.js (hunk 1 above)
Comment (michael-ciniawsky) [inline src/tasks/package.js]: We should decide upon either of, in `src` is also fine but it's just a structural thing, which can be consistent. Personally a separate test folder is cleaner imho.

**Comment by michael-ciniawsky** [comment]
Yep, for pure I/O transforming loaders/plugins this setup basically works fine, but not a default

Can I make separate repo for tests plus docs for the various setups, so we can add different test boilerplates to it while making progress ? What about `webpack-canary`, is this still a thing ?

**Comment by d3viant0ne** [comment]
- Yes, webpack-canary is still a thing though I have no idea what the status of that particular project is.

- A repo with a group of standard test setups sounds like a solid idea, go nuts.

https://github.com/webpack-contrib/webpack-defaults-testing

File: src/tasks/package.js (hunk 2 above)
Comment (d3viant0ne) [inline src/tasks/package.js]: I should have been more clear. IMO we should lint by default, unless you run into something like `webpack.build.js` where it's too restrictive to be useful.

In a case like that, just `/* eslint-disable */`

File: src/tasks/package.js (hunk 4 above)
Comment (d3viant0ne) [inline src/tasks/package.js]: So this is where I go back to perhaps running a unit & integration suite or trying to find a better way to utilize Jest in the context of Webpack ( a jest preset / plugin perhaps ).

One of Jest's biggest selling points is it's development experience. i.e...
- No compile step.
- Tests run on code changes only and only the tests that are related to that code change.
- Editor plugins for Jest execute in the background without needing to run a watch command via the command line.

While it's obviously more work, I think the end result for both maintainers & contributors would make it worth the effort.

I'll do a bit of research and see what it would take to test through Webpack without sacrificing all the best parts of Jest.

//cc @bebraw - Given your zeal for Jest, i'm sure you have an opinion or perhaps some insight here.

Comment (bebraw) [inline src/tasks/package.js]: Get Jest core involved. Here, https://github.com/webpack/webpack/issues/4499 . I think they are willing to help with whatever we might need to make it work for webpack projects.

**Comment by michael-ciniawsky** [comment]
I will revert the things discussed here later today

File: src/tasks/package.js (hunk 4 above)
Comment (michael-ciniawsky) [inline src/tasks/package.js]: Coverage for bundled tests 😛 but that could be impossible to do

**Comment by michael-ciniawsky** [comment]
What about merging start/'serve:dev' ?

**Comment by bebraw** [comment]
> What about merging start/'serve:dev' ?

Yeah, `serve:dev` feels redundant to me though it could be argued both should go away and we should replace them with a watch script. Now we have `test:watch`. Instead, we could have `watch:test` and `watch:build` to communicate the intent. Plain`start` doesn't tell much.

**Comment by bebraw** [comment]
You could also do `test:watch` and `build:watch`. That would be better namespacing I think.

**Comment by michael-ciniawsky** [comment]
Just `build:watch` and remove `start && serve:dev`?  Babel/Nodemon for watching ?

**Comment by bebraw** [comment]
> Just build:watch and remove start && serve:dev? Babel/Nodemon for watching ?

That would work for me. It's simpler and we still get the same functionality.

**Comment by michael-ciniawsky** [comment]
Ok I'm out for a few hours, updating when back in the cage 😛

**Comment by michael-ciniawsky** [comment]
@sapegin @bebraw @d3viant0ne Done, please review

**Comment by michael-ciniawsky** [comment]
@sapegin In `webpack-defaults` package.json or somewhere else aswell? I did better not touch anytjing unrelated from the beginning 😛 , otherwise please someone merge and patch please. Last call for maybe 

`build:watch => start`  => why?  => `yarn start vs. yarn run build:watch` :D

**Comment by sapegin** [comment]
@michael-ciniawsky We can keep both: `start` as a shortcut to `build:watch`. Or just `start` ;-)

**Comment by michael-ciniawsky** [comment]
kk mom :), something else ?

**Comment by michael-ciniawsky** [comment]
`build:watch` => `start`. Done, ok to merge 😛  ? I need to quicky update the `README.template` before next patch then :)

**Comment by sapegin** [comment]
Looks good to me.

**Comment by michael-ciniawsky** [comment]
@sapegin LTGM === You/I can merge it 🙃 😛  ?

**Comment by sapegin** [comment]
GitHub says that @d3viant0ne is still waiting for some changes 😱

**Comment by michael-ciniawsky** [comment]
These comments are all addressed 😉 , but kk let's wait

**Comment by michael-ciniawsky** [comment]
Anything else to do here ? Otherwise merge and release please i need this to move on with `webpack-defaults` 😛
